crates/router/src/core/admin.rs

```diff
@@ -1739,6 +1739,40 @@ pub async fn extended_card_info_toggle(
     Ok(service_api::ApplicationResponse::Json(ext_card_info_choice))
 }
 
+pub async fn connector_agnostic_mit_toggle(
+    state: AppState,
+    profile_id: &str,
+    connector_agnostic_mit_choice: admin_types::ConnectorAgnosticMitChoice,
+) -> RouterResponse<admin_types::ConnectorAgnosticMitChoice> {
+    let db = state.store.as_ref();
+
+    let business_profile = db
+        .find_business_profile_by_profile_id(profile_id)
+        .await
+        .to_not_found_response(errors::ApiErrorResponse::BusinessProfileNotFound {
+            id: profile_id.to_string(),
+        })?;
```

Comment:
after querying this, check whether this `profile` belongs to the merchant who is trying to access this endpoint